src/core.rs

```diff
@@ -0,0 +1,48 @@
+pub struct Package {
+    id: String,
+    name: String,
+    version: String,
+    date: Option<String>,
+    difficulty: Option<u8>,
+    language: Option<String>,
+    logo: Option<String>,
+    publisher: Option<String>,
+    restriciton: Option<String>,
+    rounds: Vec<Round>,
+    tags: Vec<String>,
+    info: Info
+}
+
+pub struct Info {
+    comments: String, 
+    extension: String,
+    authors: Vec<String>,
+    sources: Vec<String>
+}
+
+pub struct Round {
+    name: String,
+    variant: String, // original name "type"
```

Comment:
```suggestion
    variant: String, // fixme: original name "type"
```
и на следующих строчках тоже